index.ts

```diff
@@ -14,53 +14,45 @@ const cachedAxios = axios.create({ adapter: cache.adapter });
 
 export default async (req: NowRequest, res: NowResponse) => {
   try {
-    const format =
-      req.url && req.url.substr(req.url.length - 3) === "xml" ? "xml" : "html";
+    let format = "html";
+    if (req.url && req.url.includes(".xml")) {
+      format = "xml";
+      res.setHeader("Content-Type", "application/atom+xml");
+    }
 
-    const object = sanitizeUrl(req.url);
-    if (!object || object.username.length == 0) {
+    const username = fetchUsername(req.url);
+    if (!username) {
       throw new Error(
-        "provide page @username in url, eg. https://calmbook.page/TurismoArgentina"
+        "provide page username in url, eg. https://calmbook.page/TurismoArgentina"
       );
     }
 
     const response = await cachedAxios.get(
-      object.itemid
-        ? `https://www.facebook.com/${object.username}/posts/${
-            object.itemid
-          }?_fb_noscript=1`
-        : `https://www.facebook.com/${object.username}/posts`
+      `https://www.facebook.com/${username}/posts`
     );
 
     const $ = cheerio.load(response.data);
-    const pageInfo = buildPage($);
-    const page: Page = { username: object.username, ...pageInfo };
+    const page: Page = buildPage($);
```

Comment:
If you are going to remove property `username` from Post interface please also update `buildPage` return type. See my comment in `models/index.ts`.

```
const buildPage = ($: CheerioSelector): Omit<Page, "username">
```

Reply:
Please update `buildPage` name and return type ;)